pkg/repositories/gormimpl/common.go

```diff
@@ -99,12 +100,12 @@ func applyFilters(tx *gorm.DB, inlineFilters []common.InlineFilter, mapFilters [
 
 func applyScopedFilters(tx *gorm.DB, inlineFilters []common.InlineFilter, mapFilters []common.MapFilter) (*gorm.DB, error) {
 	for _, filter := range inlineFilters {
-		entityModel, ok := entityToModel[filter.GetEntity()]
+		tableName, ok := entityToTableName[filter.GetEntity()]
 		if !ok {
 			return nil, adminErrors.NewFlyteAdminErrorf(codes.InvalidArgument,
 				"unrecognized entity in filter expression: %v", filter.GetEntity())
 		}
-		tableName := tx.NewScope(entityModel).TableName()
```

Comment:
So it turns out this gorm method is buggy! I observed it always returned 'workflows' as the table name regardless of the entity model passed to it

Reply:
Is there a known issue for it on the gorm GH? If not, probably should file one?

Reply:
sure